File: app/src/main/java/org/tasks/location/PlaceSearchProvider.java
package org.tasks.location;

import android.os.Bundle;
import androidx.annotation.DrawableRes;
import androidx.annotation.Nullable;
import java.util.List;
import org.tasks.Callback;
import org.tasks.data.Place;

public interface PlaceSearchProvider {
  void restoreState(Bundle savedInstanceState);

  void saveState(Bundle outState);

  @DrawableRes int getAttributionRes(boolean dark);

  void search(
      String query,
      @Nullable MapPosition bias,
      Callback<List<PlaceSearchResult>> onSuccess,
      Callback<String> onError);

  void fetch(
      PlaceSearchResult placeSearchResult, Callback<Place> onSuccess, Callback<String> onError);
}

File: app/src/main/java/org/tasks/location/PlaceSearchProvider.kt
package org.tasks.location

import android.os.Bundle
import androidx.annotation.DrawableRes
import org.tasks.Callback
import org.tasks.data.Place

interface PlaceSearchProvider {
    fun restoreState(savedInstanceState: Bundle?)

    fun saveState(outState: Bundle)

    @DrawableRes
    fun getAttributionRes(dark: Boolean): Int

    suspend fun search(query: String, bias: MapPosition?): List<PlaceSearchResult>

    suspend fun fetch(placeSearchResult: PlaceSearchResult): Place
}